docs/schemaLinks.md

```diff
@@ -0,0 +1,150 @@
+# JSON Schemas and HTTP headers
+
+When writing REST interfaces, we also commonly return JSON data in response to a request.  JSON Schemas can also be
+used to provide hints about the output format we are using.
+
+The working group that writes the JSON Schema standard [has outlined two approaches for labeling outgoing responses](http://json-schema.org/latest/json-schema-core.html#anchor33).
+Both of these involve setting HTTP headers in the outgoing response, as in:
+
+    Content-Type: application/my-media-type+json; profile="http://example.com/my-hyper-schema#"
+    Link: <http://example.com/my-hyper-schema#>; rel="describedBy"
+
+This package provides two approaches to adding these headers:
+
+1. The "Schema Link Handler" grade describe on this page adds headers for both "success" and "error" messages it fields, and has the option to use a different schema URL for each.
+2. The "Schema Link Middleware" grades described on this page add headers for a single schema to all requests they are allowed to process.
+
+# Schema Link Handler
+
+This approach is better suited a wider range of scenarios, for example, when there is a chance that "success" and
+"failure" for a given endpoint might be represented by different schemas.  A common example would be an endpoint that
+returns a set of records when search terms are entered, but which returns an error message when the search terms are 
+invalid or a server error occurs.
+
+## `gpii.schema.handlerWithSchemaHeaders`
+
+An extension of [`gpii.express.handler`](https://github.com/GPII/gpii-express/blob/master/docs/handler.md) that adds
+the HTTP headers when the `sendResponse` or `sendError` invoker is called.  This grade only adds headers, it does not
+decide what success or failure looks like.  It is designed to be overlayed with any existing `gpii.express.handler`
+grade that relies on the standard `sendResponse` and `sendError` functions.  If you are already customising these in
+your handler, this grade will likely not work as expected for you.
```

Comment:
This sounds a bit discouraging to the reader. I imagine that there is every chance that the grade works, for example, unless it i) fiddles with the same headers, or ii) fully commits the response before they can be modified. We should rework this paragraph to be clearer about what the conditions really are, and also assist the reader by reworking this  implementation so that rather than functioning by overriding some invokers, instead it consists of some namespaced event handlers which the user can have some fine control over when they act.

Reply:
The latter seems like the right solution longer term, but should be discussed and planned, as it would require quite a few updates through various packages that depend on gpii-express.  I'll work on updating the wording.

Reply:
I have updated the wording to indicate the specific things to be aware of in extending the great.  Please review and comment.